### ISSUE TITLE
[Feature Request] Handle cookies in apps

### ISSUE THREAD

**Issue by azgaviperr** [Shuffle/Shuffle#923]
For a project, does matter if using Kerberos or X509, after the first request (either python or curl) I need to take back the cookie and use it to authenticate my Wallix app for the rest of the workflow. 

Could it be possible to handle the use of the cookies in the OpenAPI Apps ?

**Comment by frikky** [comment]
Yep! As discussed on Discord, I'm wondering if we should be bumping all App Creator apps to version 1.1.0 (to not break anything) and make them return all HTTP fields, as we're already doing with the HTTP apps. The body would e.g. should up in the "body" field.

**Comment by azgaviperr** [comment]
However we will need some addition to just get the full content as the cookie will need to be reused, and right now this is not possible for the apps to handle such cookies. Or maybe I am mistaken.

**Comment by azgaviperr** [comment]
We will need to test out also authentication using X509 cert, no idea how to do this with the HTTP app yet. 

My idea (and this could be decline for Kerberos auth, there is python script for kerberos authentication and harvest the session cookie):

- Authentication using X509 with the HTTP APP
-  Return of the session cookie and store it as variable. 
- The Wallix app (this is for it I need it) have a cookie field where I can setup the cookie and use the variable for value. 
- Every request during the workflow and subworkflow execution will transmit the cookie for authentication instead of the usual API KEY / JWT  / ...

**Comment by frikky** [comment]
> We will need to test out also authentication using X509 cert, no idea how to do this with the HTTP app yet.
> 
> My idea (and this could be decline for Kerberos auth, there is python script for kerberos authentication and harvest the session cookie):
> 
> * Authentication using X509 with the HTTP APP
> * Return of the session cookie and store it as variable.
> * The Wallix app (this is for it I need it) have a cookie field where I can setup the cookie and use the variable for value.
> * Every request during the workflow and subworkflow execution will transmit the cookie for authentication instead of the usual API KEY / JWT  / ...
> 
> Maybe this should become an Authentication APP. That can use other way of authentication that provide anykind of cookie, token etc for later use on other app. No idea how to setup an override option so you lose the basic auth field and get cookie or any other stuff on the open api apps

I think there's a missconception here in what differentiates "auth with a file" and "X509 auth". In my view, this is just uploading a file to an endpoint, and getting a session back which can be used in further nodes. What the file is is entirely up to the target system

We do this kind of auth already multiple places with JWT and Oauth2, but they've all had some custom creations. I think the best first step is to just add a File Upload piece to the POST request for the HTTP app
